### ISSUE TITLE
Small changes to out experience on Steam Deck / SteamOS 3

### ISSUE THREAD

**Issue by AeliusSaionji** [tkashkin/GameHub#626]
Hello!

I ended up finding GameHub because itchio linux refuses to let me download Windows games if it doesn't detect wine/proton. 😩 

Here's a non-comprehensive list of rough edges:

1. GameHub game properties window is taller than the steam deck's 1280x800. The window is nonresizeable and doesn't scroll. To access the "add to steam" button, I had to make the bottom taskbar autohide.
2. Of course, I needed the flatpak version, and the add to steam button won't work until flatpak permissions get ironed out? I'm sure you know more about this than me,
3. Gamehub (or the flatpak release) doesn't seem to work with the steam steamOS installation at the moment (GameHub just crashes shortly after starting if I point it at steam's dir)
4. Steam provides its own proton and compatibility interface. What a lot of Deck users are going to be looking for is a downloader that merely creates a *.desktop shortcut for every downloaded game (another thing itch doesn't do). No need for GameHub to provide any compatibility layer or play middleman- just need the ability to generate a .desktop shortcut for each game, to point directly to the game's exe. Then it will be a simple matter of checking off the games in steam's "add nonsteam game" menu. Or if `#2` can be worked out, it'd be nice if gamehub automatically kept the nonsteam games up to date (updating changes, removing if uninstalled, etc).
5. I see that innoextract is expected to be provided externally, but we can't really do that on SteamOS easily. The flatpak release needs access to innoextract without the user manually providing it.
6. In the meantime, I wrote a simple script to naively scrape my ~/Games folder and make .desktop shortcuts that way, https://gist.github.com/AeliusSaionji/02ca515231ea8a891a59097a631d0570 but unfortunately, itch Windows installs are stored in a folder named `game-543254` instead of just `game-name` like Linux based itch installs. It'd be nicer if these installs were always installed with a friendly name. Not just for my script, but just for general convenience.

Sorry for the long list! Thanks for the work thus far!

**Comment by isopix** [comment]
2. Gamehub's dev mentioned multiple times he sucks at making flatpaks and
is not much interested in it, and hopes somebody will help with it.

BTW. Other clients from FlatPaks (even MiniGalaxy, which is in official
FlatHub) sucks even more on SteamDeck.
Maybe it's worth switching rootfs to rw and install it native way (by
pacman), avoiding lot of headache and broken stuff (and saving much of
precious SSD space)
You will need to reinstall it every SteamOS upgrade, but is it really a
problem?

3. Anyway it natives and some windows (innoextract and DOSBox broken)
install games, they appear in 'desktop mode' in plasma menu, and you can
easily add them.
It interacts wonderfully with SteamDeck, works in both modes (game and
desktop), unlike any other clients (heroic/minigalaxy) and got beatiful
gamepad navigation.

4. Google for BoilR. Doesn't work with GameHub yet, but GameHub provides
correct .desktop files by itself
5 kwi 2022 01:49, "Aelius" ***@***.***> napisał(a):

**Comment by AeliusSaionji** [comment]
Thank you for detailing what I can do to improve my own deck experience, but much of this isn't relevant to the GameHub project.

Let's please keep this discussion about the issue at hand: what changes would be welcome to see in GameHub itself.

**Comment by isopix** [comment]
I disagree, because for SteamDeck users the most accessible way to install any software(out of steam) is to use 'Discover' in 'desktop mode'.

So the most useful improvements for SteamDeck users, would be proper 'flatpak' on 'flathub'.

Otherwise GameHub will be pretty non-existant for StemDecks users.

About GOG clients on SteamDeck, we currently have MiniGalaxy and HeroicLauncher on Flathub. Not sure a bout them, but in the past they had some problems(due to containerization).
I know that HeroicLauncher improved much and with RetroDeck having proper flatpak, GameHub would take them as example.

GameHub works somehow on SteamDeck already (like on desktop, with good gamepad support in UI) and most problem related to it, are due to containerization.
